src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -248,6 +248,34 @@ internal static class UriHelper
             return dest;
         }
 
+        internal static unsafe char[] UnescapeString(string input, int start, int end, char[] dest,
+            ref int destPosition, char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser syntax,
+            bool isQuery)
+        {
+            fixed (char* pStr = input)
+            {
+                return UnescapeString(pStr, start, end, dest, ref destPosition, rsvd1, rsvd2, rsvd3, unescapeMode,
+                    syntax, isQuery);
+            }
+        }
+
+        internal static unsafe char[] UnescapeString(char* pStr, int start, int end, char[] dest, ref int destPosition,
+            char rsvd1, char rsvd2, char rsvd3, UnescapeMode unescapeMode, UriParser syntax, bool isQuery)
+        {
+            PooledCharArray pooledArray = new PooledCharArray(dest);
+
+            UnescapeString(pStr, start, end, ref pooledArray, ref destPosition, rsvd1, rsvd2, rsvd3, unescapeMode,
+                    syntax, isQuery);
+
+            if (destPosition > dest.Length)
+            {
+                dest = new char[destPosition];
```

Comment:
>new char[destPosition]; [](start = 23, length = 23)

why isn't this getting it from the pool? any use of "new char[]" should be suspect in these codepaths now.